tests/ssg_test_suite/virt.py

```diff
@@ -142,6 +142,8 @@ def determine_ip(domain):
                 guest_agent_xml_string = ET.tostring(guest_agent_node)
                 break
         if guest_agent_xml_string:
+            if isinstance(guest_agent_xml_string, bytes):
+                guest_agent_xml_string = guest_agent_xml_string.decode('utf-8')
```

Comment:
I think you can use `encoding` argument in `tostring` that  creates this variable above. https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.tostring

Reply:
Ohhhhhhhhhhhhh I see now. From the doc:

> If encoding is "unicode", a string is returned. Otherwise a bytestring is returned.

I used `encoding='utf-8'` so the result was a bytestring NOT a string. d'oh!

Thanks @jan-cerny!